### PR TITLE
FINERACT-1905: Distribution logic fix

### PR DESCRIPTION
**Pull request by adamsaghy**
## Description

Describe the changes made and why they were made.

Ignore if these details are present on the associated [Apache Fineract JIRA ticket](https://github.com/apache/fineract/pull/1284).

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [x] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [x] Create/update unit or integration tests for verifying the changes made.

- [x] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [x] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/legacy-docs/apiLive.htm with details of any API changes

- [x] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

File: integration-tests/src/test/java/org/apache/fineract/integrationtests/common/GlobalConfigurationHelper.java [hunk 1]
```diff
@@ -677,6 +692,12 @@ public static Integer updateIsAutomaticExternalIdGenerationEnabled(final Request
         return updateEnabledFlagForGlobalConfiguration(requestSpec, responseSpec, configId, enabled);
     }
 
+    public static Integer updateChargeAccrualDate(final RequestSpecification requestSpec, final ResponseSpecification responseSpec,
```
Comment (reluxa) [inline integration-tests/src/test/java/org/apache/fineract/integrationtests/common/GlobalConfigurationHelper.java]: Isn't this a dupe of updateChargeAccrualDateConfiguration() which is right below? That also tries to update the same configId 56.

Comment (adamsaghy) [reply integration-tests/src/test/java/org/apache/fineract/integrationtests/common/GlobalConfigurationHelper.java]: You are right! I dont know how i missed this :D

File: integration-tests/src/test/java/org/apache/fineract/integrationtests/DueDateRespectiveLoanRepaymentScheduleTest.java [hunk 2]
```diff
@@ -1397,13 +1403,288 @@ public void scenario9() {
...
+            assertEquals(0.0, loanDetails.getRepaymentSchedule().getPeriods().get(1).getFeeChargesPaid());
+            assertEquals(3.65, loanDetails.getRepaymentSchedule().getPeriods().get(1).getFeeChargesOutstanding());
+            assertEquals(0.0, loanDetails.getRepaymentSchedule().getPeriods().get(1).getPenaltyChargesDue());
+            assertEquals(0.0, loanDetails.getRepaymentSchedule().getPeriods().get(1).getPenaltyChargesPaid());
+            assertEquals(0.0, loanDetails.getRepaymentSchedule().getPeriods().get(1).getPenaltyChargesOutstanding());
+            assertEquals(127.95, loanDetails.getRepaymentSchedule().getPeriods().get(1).getPrincipalDue());
+            assertEquals(125.0, loanDetails.getRepaymentSchedule().getPeriods().get(1).getPrincipalPaid());
+            assertEquals(2.95, loanDetails.getRepaymentSchedule().getPeriods().get(1).getPrincipalOutstanding());
+            assertTrue(loanDetails.getStatus().getActive());
+
```
Comment (ruchiD) [inline integration-tests/src/test/java/org/apache/fineract/integrationtests/DueDateRespectiveLoanRepaymentScheduleTest.java:1524]: Hi @adamsaghy, Do we want to check distribution of various portions(principal, fee, penalty, interest etc) for transaction also as we did for repayment schedule.
Same for rest of the test and scenario11.

**Review by reluxa — APPROVED** [review]
LGTM

**Pull request merged**